src/ir/module-splitting.cpp

```diff
@@ -196,13 +215,13 @@ TableSlotManager::Slot TableSlotManager::getSlot(Name func) {
 
   // If there are no segments yet, allocate one.
   if (activeSegment == nullptr) {
```

Comment:
This would be the best place to lazily allocate a new table if one does not already exist.

Reply:
More specifically, I would mirror the `activeSegment` mechanism with an `activeTable` mechanism that works the same way rather than using `auxiliaryTable`.

Reply:
You're right! Done.